### DISCUSSION TITLE
Is it possible to add custom claims to the JWT token?

### DISCUSSION THREAD

**Discussion by huggybaird** [fastapi-users/fastapi-users#965] [Q&A]
is it possible to add custom fields/claims to the JWT token?

For example, if i wanted to add "myrole":
{
    "user_id": "d8efb590-4e30-4e31-be91-d2db3b6bca34",
    "aud": [
        "fastapi-users:auth"
    ],
    "exp": 1650722572,
    **"myrole": "admin"**
}

**Comment by frankie567** [comment]
Hi @huggybaird!

It's possible by overloading the [`write_token` method on `JWTStrategy`](https://github.com/fastapi-users/fastapi-users/blob/e29633204d8507ffe040f5006eefe75bd86fc6a9/fastapi_users/authentication/strategy/jwt.py#L62-L66):

```py
from fastapi_users.authentication import JWTStrategy
from fastapi_users.jwt import generate_jwt

class CustomJWTStrategy(JWTStrategy):
    async def write_token(self, user: UserDB) -> str:
        data = {
            "user_id": str(user.id),
            "aud": self.token_audience,
            "myrole": "admin",  # Add your claims here
        }
        return generate_jwt(
            data, self.encode_key, self.lifetime_seconds, algorithm=self.algorithm
        )
